File: distance_matrix_API.py
import requests, json, numpy as np

# Provide origin and destination and the output is a tuple of time and distance between the origin and destination 
def get_distance(origin, destination):
    api_key = "AIzaSyCH6v-2T8u7wKKwluiX4FWUyOxcpvmTyD4"
    url = "https://maps.googleapis.com/maps/api/distancematrix/json?"+"origins="+origin+"&destinations=" +destination +"&units=imperial&key="+api_key
    payload={}
    headers = {}
    response = requests.request("GET", url, headers=headers, data=payload)
    result = json.loads(response.text)
    return result["rows"][0]["elements"][0]['duration']['value'], result["rows"][0]["elements"][0]['distance']['value']


# pass a string list of n number of location's latitude and longitude 
def createDistanceMatrix(locations):
    n = len(locations)
    D = np.zeros((n,n))
    T = np.zeros((n,n))
    for i in range(n):
        for j in range(n):
            if i!=j and j != 0:               
                T[i,j], D[i,j] = get_distance(locations[i], locations[j])
    return T, D

File: distance_matrix_API.jl
using HTTP, JSON

origin = "37.4243691,-122.1597759"
destination = "37.4258662,-122.1609591"

# Input should be latitude and longitude
# Return Time, Distance
function findTimeDistance(origin,destination)
    api_key = "AIzaSyCH6v-2T8u7wKKwluiX4FWUyOxcpvmTyD4"
    url = "https://maps.googleapis.com/maps/api/distancematrix/json?"*"origins="*origin*"&destinations=" *destination *"&key="*api_key
    r = HTTP.request("GET", url)
    body_string = (String(r.body))
    parsed_data = JSON.parse(body_string)
    return parsed_data["rows"][1]["elements"][1]["distance"]["value"] #parsed_data["rows"][1]["elements"][1]["duration"]["value"], 
end

findTimeDistance(origin, destination)


function findElevation(location)
    api_key = "AIzaSyCH6v-2T8u7wKKwluiX4FWUyOxcpvmTyD4"
    elevation_url = "https://maps.googleapis.com/maps/api/elevation/json?locations="*location*"&key="*api_key
    r = HTTP.request("GET", elevation_url)
    body_string = (String(r.body))
    parsed_data = JSON.parse(body_string)
    return parsed_data["results"][1]["elevation"]
end

findElevation(origin)